src/tag.c

```diff
@@ -289,6 +289,9 @@ do_tag(
     static char_u	**matches = NULL;
     static int		flags;
 
+    if (!check_can_set_curbuf_forceit(forceit))
+        return FALSE;
```

Comment:
and here please

Reply:
This note and https://github.com/vim/vim/pull/13903#discussion_r1501385399 are interrelated. If the other block and this block are removed then the following tests will fail

Test_normal_ctrl_w_g_ctrl_square_bracket_right
Test_normal_g_square_bracket_right
Test_normal_v_ctrl_square_bracket_right
Test_normal_v_g_ctrl_square_bracket_right
Test_tNext
Test_tprevious

However if we add this block back in then the above tests pass again. Which further makes me think that we can remove https://github.com/vim/vim/pull/13903#discussion_r1501385399 and keep this block. That said, this block is used in all of the above tests. Not sure why the automated coverage is saying otherwise.

Reply:
okay, then please do so and consolidate

Reply:
Done!